flink-rpc/flink-rpc-akka/src/main/java/org/apache/flink/runtime/rpc/pekko/CustomSSLEngineProvider.java

```diff
@@ -59,13 +68,35 @@ public TrustManager[] trustManagers() {
                                     .fingerprints(sslCertFingerprints)
                                     .build();
 
-            trustManagerFactory.init(loadKeystore(sslTrustStore, sslTrustStorePassword));
+            trustManagerFactory.init(
+                    loadKeystore(sslTrustStore, sslTrustStorePassword, sslTrustStoreType));
             return trustManagerFactory.getTrustManagers();
-        } catch (GeneralSecurityException e) {
+        } catch (GeneralSecurityException | IOException e) {
             // replicate exception handling from SSLEngineProvider
             throw new RemoteTransportException(
                     "Server SSL connection could not be established because SSL context could not be constructed",
                     e);
         }
     }
+
+    @Override
+    public KeyStore loadKeystore(String filename, String password) {
+        try {
+            return loadKeystore(filename, password, sslKeyStoreType);
+        } catch (IOException | GeneralSecurityException e) {
+            throw new RemoteTransportException(
+                    "Server SSL connection could not be established because SSL context could not be constructed",
```

Comment:
Here we can be more specific: "Server SSL connection could not be established because keystore could not be loaded"